### PR TITLE
Remove old dev server docs

### LINKED ISSUE #12767: [BUG] No module named 'bokeh.server.tests'
### Software versions

Python version      :  3.10.8 | packaged by conda-forge | (main, Nov 22 2022, 08:26:04) [GCC 10.4.0]
IPython version     :  8.8.0
Tornado version     :  6.2
Bokeh version       :  3.1.0.dev1+19.g0e72ad3c
BokehJS static path :  ..../git/bokeh/src/bokeh/server/static
node.js version     :  v16.17.1
npm version         :  8.19.3
Operating system    :  Linux-5.14.0-1056-oem-x86_64-with-glibc2.31

### Browser name and version

_No response_

### Jupyter notebook / Jupyter Lab version

_No response_

### Expected behavior

following the dev. guide, in the server.html page:

Testing
To test client-server functionality, use the utilities in bokeh.server.tests.utils.

### Observed behavior

ModuleNotFoundError: No module named 'bokeh.server.tests'

### Example code

```Python
import bokeh.server.tests.utils
```

### Stack traceback or browser console output

_No response_

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #12767 
- [x] release document entry (if new feature or API change)

@mattpap there was an old "datepicker" ref that was broken, not sure why docs built in the old PR to be honest. As long as that is updated, can you confirm whether the list of widgets that can have tooltips is still accurate (do the new pickers also support?)

### CODE REVIEW

File: docs/bokeh/source/docs/user_guide/interaction/tooltips.rst [hunk 1]
```diff
@@ -71,7 +71,7 @@ Currently, the following input widgets support tooltips directly:
 
 * :ref:`ug_interaction_widgets_examples_autocompleteinput`
 * :ref:`ug_interaction_widgets_examples_colorpicker`
-* :ref:`ug_interaction_widgets_examples_datepicker`
+* :ref:`ug_interaction_widgets_examples_date_picker`
```
Comment (mattpap) [inline docs/bokeh/source/docs/user_guide/interaction/tooltips.rst:74]: I would expect the original to fail compilation. I wonder how many more refs are broken across the docs.

**Comment by mattpap** [comment]
>  As long as that is updated, can you confirm whether the list of widgets that can have tooltips is still accurate (do the new pickers also support?)

All input widgets (give or take widgets with a label) support tooltips. So, all new pickers do support them.

File: docs/bokeh/source/docs/user_guide/interaction/tooltips.rst (hunk 1 above)
Comment (bryevdv) [reply docs/bokeh/source/docs/user_guide/interaction/tooltips.rst:74]: It failed for me locally, which is how I found it. Not sure why it didn't in CI

**Comment by bryevdv** [comment]
I'll plan to merge this tomorrow without objection.

**Review by mattpap — APPROVED** [review]

**Comment by bryevdv** [comment]
Just noting I also decided that this should occur for pre-releases, since their URL prefix is also re-used. I added this in direct commit https://github.com/bokeh/bokeh/commit/67e2eba9bc30729a3097d358c7536f67230c7326

**Pull request merged**